### PR TITLE
[fix]: [CI-4345]: namespace and serviceaccount whitespace

### PR DESCRIPTION
**Pull request by smjt-h**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32865)

### CODE REVIEW

**Review by aman-harness — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32865#-:-N1PJq-eA9HmO9wluFiT:bnfp4nl)* status: :shipit: complete! ✓

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java [hunk 1]
```diff
@@ -126,6 +124,17 @@ public K8sTaskExecutionResponse executeTaskInternal(
     PodParams podParams = cik8InitializeTaskParams.getCik8PodParams();
     String namespace = podParams.getNamespace();
     String podName = podParams.getName();
+    String serviceAccountName = podParams.getServiceAccountName();
+
+    if (namespace != null) {
```
Comment (shubham149) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:129]: @smjt-h Removal of spaces in namespace should be done in ci-manager and not the delegate side. Can you fix this?

Comment (shubham149) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:129]: Same for service account name

Comment (shubham149) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:129]: Also, use trim to remove leading and trailing spaces https://www.tutorialspoint.com/trim-a-string-in-java-to-remove-leading-and-trailing-spaces#:~:text=To%20remove%20leading%20and%20trailing%20spaces%20in%20Java%2C%20use%20the,leading%20or%20trailing%20white%20space.

**Pull request merged**